src/Bundler.js

```diff
@@ -643,7 +643,9 @@ class Bundler extends EventEmitter {
   }
 
   async serve(port = 1234, https = false) {
-    this.server = await Server.serve(this, port, https);
+    if (this.options.target === 'browser') {
```

Comment:
I think we should move this check into the CLI specifically. If someone is using the Parcel API, and they call `serve`, we should always start a server even if the target is node. In the CLI, we can detect the target and just call `bundle` instead of `serve` there.